packages/ts/generator-typescript-plugin-model/src/ModelSchemaProcessor.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  isNullableSchema,
+  Schema,
+  decomposeSchema,
+  isComposedSchema,
+  NonComposedSchema,
+  isReferenceSchema,
+  isNumberSchema,
+  isIntegerSchema,
+  isStringSchema,
+  isMapSchema,
+  isArraySchema,
+  isBooleanSchema,
+  convertReferenceSchemaToPath,
+  convertReferenceSchemaToSpecifier,
+  ReferenceSchema,
+  ArraySchema,
+  MapSchema,
+  NonComposedRegularSchema,
+  isNonComposedRegularSchema,
+} from '@vaadin/generator-typescript-core/Schema.js';
+import PluginError from '@vaadin/generator-typescript-utils/PluginError.js';
+import type DependencyManager from '@vaadin/generator-typescript-utils/dependencies/DependencyManager';
+import type { ArrayLiteralExpression, Expression, Identifier, TypeNode } from 'typescript';
+import ts from 'typescript';
+
+type AnnotatedSchema = NonComposedRegularSchema & Readonly<{ 'x-annotations': ReadonlyArray<string> }>;
+
+function isAnnotatedSchema(schema: Schema): schema is AnnotatedSchema {
+  return isNonComposedRegularSchema(schema) && 'x-annotations' in schema;
+}
+
+type AnnotationPrimitiveArgument = boolean | number | string;
+type AnnotationArgument = AnnotationPrimitiveArgument | AnnotationNamedArguments;
+interface AnnotationNamedArguments {
```

Comment:
Let's have `type` here and below to be more consistent across the generator-related codebase.